### DISCUSSION TITLE
Get Rid of the New Custom Fetch

### DISCUSSION THREAD

**Discussion by jdgamble555** [vercel/next.js#42775] [Ideas]
### Describe the feature you'd like to request

While I love the `app` directory structure in V13+, I am not happy about moving the ISR and Caching into the `fetch` api.  This is extremely limiting, and creates MORE work, not less for the developers.

Image I am using Firebase, Supabase, Apollo, or any package with a custom use-hook with fetch already in it.  I will NOT have access to the native fetch used in that package. The only work around will be to create an extra file that the server file has to fetch.  That means I will be fetching another file on the server, from the server.  This is pointless extra work, when before I could just use `getServerSideProps` or `getStaticProps` to return data I retrieve however I wanted.  This *was* a problem in SvelteKit as well, and should be revisited.

- [SvelteKit Github Same Problem](https://github.com/sveltejs/kit/issues/3729)
- [My article looking for a solution](https://dev.to/jdgamble555/passing-state-from-the-server-to-the-browser-in-sveltekit-37co)

We don't want to be LIMITED to using fetch in order to Cache or use ISR. Period.

### Describe the solution you'd like

Honestly, sticking with passing the data just like `getServerSideProps` and `getStaticProps` seemed to work just fine, although only one function might be better (as well as `getStaticPaths`, but for a different purpose).

### Describe alternatives you've considered

Stick with `getServerSideProps` until you get rid of it, but it looks like we won't have a choice in the future.

J

**Comment by ijjk** [comment]
Hi, fetch isn't required for `app` directory it just allows more fine grain caching control you can still achieve the same behavior as `getServerSideProps`/`getStaticProps` by having a single data fetching method in your page and configuring `export const revalidate`. 

Note we are still investigating improving the fine grain control without fetch as well.

**Comment by jdgamble555** [comment]
Ok great. I would say as long as you don't get rid of that feature, no problem. No point in creating an extra route.

https://beta.nextjs.org/docs/data-fetching/caching#segment-level-caching

Thanks,
J
